### PR TITLE
Rename BitmapLoader into DefaultAsyncRetriever

### PR DESCRIPTION
**Pull request by friedger**
BitmapLoader is an asynchronous Retriever, so lets call it so.

Separate Retriever.Callback and AsyncRetriever.Callback

### CODE REVIEW

File: core/src/main/java/com/novoda/pxfetcher/AsyncRetriever.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.novoda.pxfetcher;
+
+import com.novoda.pxfetcher.task.Retriever;
+import com.novoda.pxfetcher.task.TagWrapper;
+
+public interface AsyncRetriever {
+    void load(TagWrapper tagWrapper, Callback callback);
+
+    interface Callback extends Retriever.Callback {
+        void onStart();
```
Comment (blundell) [inline core/src/main/java/com/novoda/pxfetcher/AsyncRetriever.java]: I would add the overriden methods here as well with the @Override annotation, it lets you easily see the whole interface

File: core/src/main/java/com/novoda/pxfetcher/DefaultAsyncRetriever.java [hunk 2]
```diff
@@ -2,16 +2,20 @@
...
 import android.os.AsyncTask;
 
-import com.novoda.pxfetcher.task.*;
+import com.novoda.pxfetcher.task.Result;
+import com.novoda.pxfetcher.task.Retriever;
+import com.novoda.pxfetcher.task.RetrieverAsyncTask;
+import com.novoda.pxfetcher.task.TagWrapper;
 
-public class BitmapLoader {
+public class DefaultAsyncRetriever implements AsyncRetriever {
```
Comment (blundell) [inline core/src/main/java/com/novoda/pxfetcher/DefaultAsyncRetriever.java]: what does default mean? Add javadoc OR rename to something that matches it's behaviour

Comment (ouchadam) [inline core/src/main/java/com/novoda/pxfetcher/DefaultAsyncRetriever.java]: wouldn't `Retriever` be the interface and `Async` the implementation?

Comment (friedger) [inline core/src/main/java/com/novoda/pxfetcher/DefaultAsyncRetriever.java]: @ouchadam It is a diffferent interface.

File: core/src/main/java/com/novoda/pxfetcher/AsyncRetriever.java (hunk 1 above)
Comment (blundell) [inline core/src/main/java/com/novoda/pxfetcher/AsyncRetriever.java]: :+1:

File: core/src/main/java/com/novoda/pxfetcher/AsyncTaskRetriever.java [hunk 3]
```diff
@@ -2,16 +2,20 @@
...
 import android.os.AsyncTask;
 
-import com.novoda.pxfetcher.task.*;
+import com.novoda.pxfetcher.task.Result;
+import com.novoda.pxfetcher.task.Retriever;
+import com.novoda.pxfetcher.task.RetrieverAsyncTask;
+import com.novoda.pxfetcher.task.TagWrapper;
 
-public class BitmapLoader {
+public class AsyncTaskRetriever implements AsyncRetriever {
```
Comment (blundell) [inline core/src/main/java/com/novoda/pxfetcher/AsyncTaskRetriever.java]: `public class TaskAsyncRetriever implements AsyncRetriever`

Comment (blundell) [inline core/src/main/java/com/novoda/pxfetcher/AsyncTaskRetriever.java]: this change and :shipit: !

Comment (friedger) [inline core/src/main/java/com/novoda/pxfetcher/AsyncTaskRetriever.java]: Disagree! The emphasis is on AsyncTask, not on AsyncRetriever. If AsyncTaskReceiver is not appropriate then it should be AsyncTaskAsyncRetriever which looks too much Async.

**Comment by friedger** [comment]
Close until #214 is merged

**Pull request closed without merging**